src/vaadin-text-field-mixin.html

```diff
@@ -680,10 +717,17 @@
       }
     }
 
-    _getActiveErrorId(invalid, errorMessage, errorId) {
-      this._setOrToggleAttribute('aria-describedby',
-        (errorMessage && invalid ? errorId : undefined),
-        this.focusElement);
+    _getActiveErrorId(invalid, errorMessage, errorId, helperText, helperTextId, hasSlottedHelper) {
+      let ids = '';
+      const hasHelper = helperText || hasSlottedHelper;
+      if (hasHelper) {
+        ids += helperTextId;
+      }
+      if (errorMessage && invalid) {
+        ids += hasHelper ? ' ' : '';
```

Comment:
nit: building a space separated string like this is a bit clumsy. Rather push to an array of strings and then use `ids.join(' ')` to form the space separated string.

Reply:
Done